### PR TITLE
Applying the parse framework to the Footprint store class

### PR DESCRIPTION
**Pull request by rt17603**
* **Summary of changes** (Bug fix, feature, docs update, ...)

At the moment, for a subset of available data types this uses functions within the `openghg.standardise` submodule to create `parse_*` functions. This means that different input types can be specified and standardised using a `source_format` input.

This PR adds this functionality to the `Footprint` type (through the `standardise_footprint` input) and creates a new parser function called `parse_acrg_org`.

This is included in preparation for adding in new footprint parser functionality for the new footprint format - #905 .

* **Please check if the PR fulfills these requirements**

- [x] Related to #905 but doesn't close this
- [x] [Tests added and passed](https://docs.openghg.org/development/python_devel.html#testing) if fixing a bug or adding a new feature
   -  No new tests added as this is refactoring code.
- [x] All code checks passing - `black --line-length 110` run over code, `mypy` and `flake8` not showing any errors
- [x] Added an entry in the latest `CHANGELOG.md` file if fixing a bug or adding a new feature

### CODE REVIEW

**Review by gareth-j — CHANGES_REQUESTED** [review]
Just a few changes around handling species and some commented code

**Comment by rt17603** [comment]
Thanks for the review - oddly I feel like I had thought I had already addressed some of the changes you pointed out (may have been on PR #918 which branches off this). Will make sure those changes are addressed.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,8 +5,6 @@ All notable changes to OpenGHG will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [Unreleased](https://github.com/openghg/openghg/compare/0.7.0...HEAD)
```
[the remark this answers is not in the document]
Comment (rt17603) [reply CHANGELOG.md:8]: With the 0.8.0 line below would we expect to keep both?

File: openghg/standardise/footprint/_acrg_org.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+
+def parse_acrg_org(
+    filepath: Union[Path, List[Path]],
+    site: str,
+    domain: str,
+    model: str,
+    inlet: str,
+    metmodel: Optional[str] = None,
+    species: Optional[str] = None,
```
[the remark this answers is not in the document]
Comment (rt17603) [reply openghg/standardise/footprint/_acrg_org.py]: Updated `species` type to `str`

File: openghg/standardise/footprint/_acrg_org.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    metadata["data_type"] = "footprints"
+    metadata["site"] = site
+    metadata["domain"] = domain
+    metadata["model"] = model
+
+    # Include both inlet and height keywords for backwards compatability
+    metadata["inlet"] = inlet
+    metadata["height"] = inlet
+
+    if species is not None:
```
[the remark this answers is not in the document]
Comment (rt17603) [reply openghg/standardise/footprint/_acrg_org.py]: Updated to always store species value in metadata.

File: openghg/standardise/footprint/_acrg_org.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        dict: Dictionary of data
+    """
+
+    xr_open_fn, filepath = check_function_open_nc(filepath)
+
+    with xr_open_fn(filepath).chunk(chunks) as fp_data:
+        if chunks:
+            logger.info(f"Rechunking with chunks={chunks}")
+
+    # # Load this into memory
```
[the remark this answers is not in the document]
Comment (rt17603) [reply openghg/standardise/footprint/_acrg_org.py]: Done.

File: openghg/standardise/footprint/_acrg_org.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        species: Species name. Only needed if footprint is for a specific species e.g. co2 (and not inert)
+        network: Network name
+        period: Period of measurements. Only needed if this can not be inferred from the time coords
+        continuous: Whether time stamps have to be continuous.
+        high_spatial_resolution : Indicate footprints include both a low and high spatial resolution.
+        high_time_resolution: Indicate footprints are high time resolution (include H_back dimension)
+                        Note this will be set to True automatically if species="co2" (Carbon Dioxide).
+        short_lifetime: Indicate footprint is for a short-lived species. Needs species input.
+                        Note this will be set to True if species has an associated lifetime.
+        chunks: Chunk size to use when parsing NetCDF, useful for large datasets.
```
[the remark this answers is not in the document]
Comment (rt17603) [reply openghg/standardise/footprint/_acrg_org.py:50]: Removed note about auto - did we want to include any other details though about how chunks are chosen?

File: openghg/standardise/footprint/_acrg_org.py [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+    inlet: str,
+    metmodel: Optional[str] = None,
+    species: Optional[str] = None,
+    network: Optional[str] = None,
+    period: Optional[Union[str, tuple]] = None,
+    continuous: bool = True,
+    high_spatial_resolution: bool = False,
+    high_time_resolution: bool = False,
+    short_lifetime: bool = False,
+    chunks: Union[int, Dict, Literal["auto"], None] = None,
```
[the remark this answers is not in the document]
Comment (rt17603) [reply openghg/standardise/footprint/_acrg_org.py]: Updated - this had been updated in the `Footprint` class but change had not made it to this parse function - now done.

**Comment by rt17603** [comment]
Just a question on the Unreleased line and 0.8.0 in the CHANGELOG.mg - what is the typical format we are using here?

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (gareth-j) [reply CHANGELOG.md:8]: Yeah I think the Unreleased line should be kept and the release line should be added when we're about to do the release.

[the remark this answers is not in the document]
Comment (gareth-j) [reply CHANGELOG.md:8]: Then we just bump the version in the Unreleased URL

**Review by gareth-j — APPROVED** [review]
Looks good!

**Comment by gareth-j** [comment]
> Just a question on the Unreleased line and 0.8.0 in the CHANGELOG.mg - what is the typical format we are using here?

Previously we've just added all the changes under Unreleased and then I've added the release number just before doing the release. Then the URL used for the unreleased changes is updated with the new version number.

File: openghg/standardise/footprint/_acrg_org.py [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+        species: Species name. For a long-lived species this should be "inert".
+        network: Network name
+        period: Period of measurements. Only needed if this can not be inferred from the time coords
+        continuous: Whether time stamps have to be continuous.
+        high_spatial_resolution : Indicate footprints include both a low and high spatial resolution.
+        high_time_resolution: Indicate footprints are high time resolution (include H_back dimension)
+                        Note this will be set to True automatically if species="co2" (Carbon Dioxide).
+        short_lifetime: Indicate footprint is for a short-lived species. Needs species input.
+                        Note this will be set to True if species has an associated lifetime.
+        chunks: Chunk size to use when parsing NetCDF, useful for large datasets.
```
Comment (gareth-j) [inline openghg/standardise/footprint/_acrg_org.py:50]:
```suggestion
       chunks: Chunk schema to use when storing data the NetCDF. It expects a dictionary of dimension name and chunk size, for example {"time": 100}. If None then a chunking schema will be set automatically by OpenGHG.
```

File: openghg/store/_footprints.py [hunk 8]
```diff
@@ -224,6 +224,9 @@ def read_file(
             network: Network name
             period: Period of measurements. Only needed if this can not be inferred from the time coords
             continuous: Whether time stamps have to be continuous.
+            chunks: Chunk size to use when parsing NetCDF, useful for large datasets.
```
Comment (gareth-j) [inline openghg/store/_footprints.py:227]:
```suggestion
            chunks: Chunk schema to use when storing data the NetCDF. It expects a dictionary of dimension name and chunk size, for example {"time": 100}. If None then a chunking schema will be set automatically by OpenGHG.
```

**Review by gareth-j — APPROVED** [review]
I think the suggested changes have been added now.

File: CHANGELOG.md [hunk 9]
```diff
@@ -5,8 +5,6 @@ All notable changes to OpenGHG will be documented in this file.
...
-
 ## 0.8.0 - xxxx-xx-xx
```
Comment (rt17603) [inline CHANGELOG.md:8]:
```suggestion
## [Unreleased](https://github.com/openghg/openghg/compare/0.7.0...HEAD)
```

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rt17603) [reply CHANGELOG.md:8]: Yeah I'm not sure when the 0.8.0 line got added (may have just been in new_object_store). I've added the line back in and removed the 0.8.0 line for now.

File: openghg/standardise/footprint/_acrg_org.py (hunk 7 above)
Comment (rt17603) [reply openghg/standardise/footprint/_acrg_org.py]: Ok I added those changes locally and committed.

File: openghg/store/_footprints.py (hunk 8 above)
Comment (rt17603) [reply openghg/store/_footprints.py]: Also added locally and committed.

**Pull request merged**